packages/manager/apps/pci-instances/src/data/hooks/instance/useInstances.ts

```diff
@@ -69,6 +71,25 @@ export const updateDeletedInstanceStatus = (
   );
 };
 
+export const getInstanceNameById = (
+  id: string | null,
+  queryClient: QueryClient,
+): string | undefined => {
+  if (!id) return undefined;
+
+  const data = queryClient.getQueriesData<InfiniteData<TInstanceDto[], number>>(
+    {
+      predicate: (query: Query) => query.queryKey.includes('list'),
+    },
+  );
+  return data.reduce((acc, [, result]) => {
+    if (acc.length) return acc;
+    if (result)
+      return result.pages.flat().find((elt) => elt.id === id)?.name ?? acc;
+    return acc;
+  }, '');
```

Comment:
Could this be handled with find only instead ?

Reply:
@SimonChaumet 
Mmmh, yep we could do something like that: 

```ts
  return (
    data
      .flatMap(([, result]) => result?.pages ?? [])
      .flat()
      .find((elt) => elt.id === id)?.name ?? ''
  );
  ```

I'am just a bit annoyed by applying multiple flat() functions, what do you think ?

Reply:
Will see later